File: Soe.Common/Util/SettingsUtility.cs
using System;
using System.Collections.Generic;

namespace SoftOne.Soe.Common.Util
{
    public static class SettingsUtility
    {
        #region User Settings

        public static String GetStringUserSetting(Dictionary<int, object> settings, UserSettingType type, string defaultValue = "")
        {
            try
            {
                return settings.ContainsKey((int)type) ? settings[(int)type].ToString() : defaultValue;
            }
            catch
            {
                return defaultValue;
            }
        }

        public static int GetIntUserSetting(Dictionary<int, object> settings, UserSettingType type, int defaultValue = 0)
        {
            try
            {
                return settings.ContainsKey((int)type) ? Int32.Parse(settings[(int)type].ToString()) : defaultValue;
            }
            catch
            {
                return defaultValue;
            }
        }

        public static bool GetBoolUserSetting(Dictionary<int, object> settings, UserSettingType type, bool defaultValue = false)
        {
            try
            {
                return settings.ContainsKey((int)type) ? Boolean.Parse(settings[(int)type].ToString()) : defaultValue;
            }
            catch
            {
                return defaultValue;
            }
        }

        public static DateTime GetDateUserSetting(Dictionary<int, object> settings, UserSettingType type, DateTime defaultValue)
        {
            try
            {
                return settings.ContainsKey((int)type) ? DateTime.Parse(settings[(int)type].ToString()) : defaultValue;
            }
            catch
            {
                return defaultValue;
            }
        }

        #endregion

        #region Company Settings

        public static String GetStringCompanySetting(Dictionary<int, object> settings, CompanySettingType type, string defaultValue = "")
        {
            try
            {
                return settings.ContainsKey((int)type) ? settings[(int)type].ToString() : defaultValue;
            }
            catch
            {
                return defaultValue;
            }
        }

        public static int GetIntCompanySetting(Dictionary<int, object> settings, CompanySettingType type, int defaultValue = 0, bool allowZero = true)
        {
            try
            {
                int value = defaultValue;
                if (settings.ContainsKey((int)type))
                {
                    foreach (var key in settings)
                    {
                        if (key.Key == (int)type)
                        {
                            value = Convert.ToInt32(key.Value);
                            break;
                        }
                    }
                }

                if (!allowZero && value == 0)
                    value = defaultValue;

                return value;
            }
            catch
            {
                return defaultValue;
            }
        }

        public static decimal GetDecimalCompanySetting(Dictionary<int, object> settings, CompanySettingType type, decimal defaultValue = 0)
        {
            try
            {
                return settings.ContainsKey((int)type) ? Decimal.Parse(settings[(int)type].ToString()) : defaultValue;
            }
            catch
            {
                return defaultValue;
            }
        }

        public static bool GetBoolCompanySetting(Dictionary<int, object> settings, CompanySettingType type, bool defaultValue = false)
        {
            try
            {
                return settings.ContainsKey((int)type) ? Boolean.Parse(settings[(int)type].ToString()) : defaultValue;
            }
            catch
            {
                return defaultValue;
            }
        }

        public static DateTime GetDateCompanySetting(Dictionary<int, object> settings, CompanySettingType type, DateTime defaultValue)
        {
            try
            {
                return settings.ContainsKey((int)type) ? DateTime.Parse(settings[(int)type].ToString()) : defaultValue;
            }
            catch
            {
                return defaultValue;
            }
        }

        #endregion
    }
}

File: Soe.Angular/TypeScript/Util/SettingsUtility.ts
import { UserSettingType, CompanySettingType, LicenseSettingType } from "../Util/CommonEnumerations";

export class SettingsUtility {

    // User setting
    public static getStringUserSetting(settings: any[], type: UserSettingType, defaultValue: string = ''): string {
        return settings[type] !== undefined ? settings[type] : defaultValue;
    }

    public static getIntUserSetting(settings: any[], type: UserSettingType, defaultValue: number = 0, allowZero: boolean = true): number {
        var value: number = settings[type] !== undefined ? settings[type] : defaultValue;

        if (value === 0 && !allowZero)
            value = defaultValue;

        return value;
    }

    public static getBoolUserSetting(settings: any[], type: UserSettingType, defaultValue: boolean = false): boolean {
        return settings[type] !== undefined ? settings[type] : defaultValue;
    }

    // Company setting
    public static getStringCompanySetting(settings: any[], type: CompanySettingType, defaultValue: string = ''): string {
        return settings[type] !== undefined ? settings[type] : defaultValue;
    }

    public static getIntCompanySetting(settings: any[], type: CompanySettingType, defaultValue: number = 0, allowZero: boolean = true): number {
        var value: number = settings[type] !== undefined ? settings[type] : defaultValue;

        if (value === 0 && !allowZero)
            value = defaultValue;

        return value;
    }

    public static getBoolCompanySetting(settings: any[], type: CompanySettingType, defaultValue: boolean = false): boolean {
        return settings[type] !== undefined ? settings[type] : defaultValue;
    }

    // License setting
    public static getStringLicenseSetting(settings: any[], type: LicenseSettingType, defaultValue: string = ''): string {
        return settings[type] !== undefined ? settings[type] : defaultValue;
    }

    public static getIntLicenseSetting(settings: any[], type: LicenseSettingType, defaultValue: number = 0, allowZero: boolean = true): number {
        var value: number = settings[type] !== undefined ? settings[type] : defaultValue;

        if (value === 0 && !allowZero)
            value = defaultValue;

        return value;
    }

    public static getBoolLicenseSetting(settings: any[], type: LicenseSettingType, defaultValue: boolean = false): boolean {
        return settings[type] !== undefined ? settings[type] : defaultValue;
    }
}
